movement/movement.c

```diff
@@ -533,15 +537,47 @@ bool app_loop(void) {
             stringToUTF8(tx, s, len);
             return s;
         });
-        memcpy(line, received_data, min(255, strlen(received_data)));
+        serial_buffer_index = strlen(received_data);
+        memcpy(serial_buffer, received_data, min(254, serial_buffer_index));
+        serial_buffer[serial_buffer_index] = '\0';
         free(received_data);
         EM_ASM({
             tx = "";
         });
+        if (serial_buffer[0] != '\0') {
+            printf("%s", serial_buffer);
+            filesystem_process_command(serial_buffer);
+            serial_buffer[0] = '\0';
+        }
 #else
-        read(0, line, 256);
+        char *read_loc = &serial_buffer[serial_buffer_index];
```

Comment:
This elaborate buffering dance handles arbitrary amounts of input at a time. I've been using:

```picocom /dev/ttyACM0 --imap lfcrlf```